# Titanic Dataset - My First EDA

This is my first Exploratory Data Analysis assignment. I'm using the Titanic dataset from Kaggle to practice pandas and understand data better.

In [3]:
import pandas as pd

# Load the Titanic dataset
df = pd.read_csv('Titanic-Dataset.csv')

# Show first 5 rows
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


In [5]:
print("Dataset shape:", df.shape)
print(f"Rows: {df.shape[0]}, Columns: {df.shape[1]}")
df.describe()

Dataset shape: (891, 12)
Rows: 891, Columns: 12


,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [6]:
print("Missing values in each column:")
print(df.isnull().sum())

Missing values in each column:
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


In [7]:
print("Column data types:")
print(df.dtypes)

numerical_cols = df.select_dtypes(include=['int64', 'float64']).columns.tolist()
categorical_cols = df.select_dtypes(include=['object']).columns.tolist()

print("\nNumerical columns:", numerical_cols)
print("Categorical columns:", categorical_cols)

Column data types:
PassengerId      int64
Survived         int64
Pclass           int64
Name               str
Sex                str
Age            float64
SibSp            int64
Parch            int64
Ticket             str
Fare           float64
Cabin              str
Embarked           str
dtype: object

Numerical columns: ['PassengerId', 'Survived', 'Pclass', 'Age', 'SibSp', 'Parch', 'Fare']
Categorical columns: ['Name', 'Sex', 'Ticket', 'Cabin', 'Embarked']


C:\Users\pc\AppData\Local\Temp\ipykernel_20132\2826896482.py:5: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  categorical_cols = df.select_dtypes(include=['object']).columns.tolist()


In [8]:
print("Survived values:", df['Survived'].unique())
print("Sex values:", df['Sex'].unique())
print("Embarked values:", df['Embarked'].unique())

print("\nMissing values:")
print(f"Age: {df['Age'].isnull().sum()} missing")
print(f"Cabin: {df['Cabin'].isnull().sum()} missing")
print(f"Embarked: {df['Embarked'].isnull().sum()} missing")

Survived values: [0 1]
Sex values: <StringArray>
['male', 'female']
Length: 2, dtype: str
Embarked values: <StringArray>
['S', 'C', 'Q', nan]
Length: 4, dtype: str

Missing values:
Age: 177 missing
Cabin: 687 missing
Embarked: 2 missing


In [9]:
survival_counts = df['Survived'].value_counts()
print("Survival counts:")
print(survival_counts)
print(f"\nSurvival rate: {(survival_counts[1]/len(df)*100):.2f}%")

Survival counts:
Survived
0    549
1    342
Name: count, dtype: int64

Survival rate: 38.38%


## My Observations

After doing my first Exploratory Data Analysis on the Titanic dataset, here's what I found:

**Dataset Size:**
- The dataset has 891 rows and 12 columns
- 38.38% of passengers survived (342 survived, 549 did not survive)

**Missing Values I Found:**
- Age has 177 missing values - this is a significant amount and needs to be handled
- Cabin has 687 missing values - more than 75% of the data is missing here, which is a major issue
- Embarked has only 2 missing values - this is easy to fix

**Column Types:**
- Numerical columns: Age, Fare, SibSp, Parch (these are actual numbers that make mathematical sense)
- Categorical columns: Name, Sex, Ticket, Cabin, Embarked (these are text or categories)
- Note: Survived is stored as numbers (0 and 1) but it's actually categorical - it represents two categories (survived or didn't survive)

**What I Think Needs Cleaning Before Machine Learning:**
1. Cabin column - with 687 missing values out of 891 rows, this column has too much missing data. I might need to drop this column entirely because filling that many missing values wouldn't be reliable.

2. Age column - 177 missing values is a lot but not as bad as Cabin. I need to decide how to fill these missing values. Maybe I could use the average age or median age.

3. Embarked column - only 2 missing values, I can easily fill these with the most common value (mode).

4. PassengerId and Ticket columns - these seem like unique identifiers and probably won't help with predictions. I might drop these later.

**What I Learned:**
- Real data is never perfect - there are always missing values and issues to deal with
- Not all numbers are numerical data - some numbers represent categories (like Survived)
- Data cleaning is really important before doing any machine learning
- Different columns need different approaches for handling missing values
- EDA helps me understand the data before I try to build any models

This is my first time doing EDA and I'm still learning, but I can see why data scientists spend so much time on data cleaning. The next step for me would be to learn how to handle these missing values properly and prepare the data for machine learning.